libs/state/src/lib/rx-state.service.ts

```diff
@@ -109,13 +108,8 @@ export class RxState<State extends object>
    */
   constructor() {
     this.subscription.add(this.subscribe());
-  }
 
-  /**
-   * @internal
-   */
-  ngOnDestroy(): void {
-    this.subscription.unsubscribe();
+    this.destroyRef.onDestroy(() => this.subscription.unsubscribe());
```

Comment:
If you only do this, we could maybe `inject(DestroyRef)` inside the constructor and drop the `destroyRef` prop.